lib/users.js

```diff
@@ -349,11 +350,11 @@ Users.prototype.list = function (search, options, callback) {
     options = {};
   }
 
-  var opts = {
-    scope: 'sub',
-    filter: search ? nconf.get('LDAP_SEARCH_QUERY').replace(/\{0\}/ig, search) : nconf.get('LDAP_SEARCH_ALL_QUERY')
-  };
+  const filterString = search ? nconf.get('LDAP_SEARCH_QUERY').replace(/\{0\}/ig, search) : nconf.get('LDAP_SEARCH_ALL_QUERY');
+  const filter = parseLdapFilter(filterString);
+  if (!filter.isValid) return callback(null, []);  // Search query is not valid, return empty results
```

Comment:
This is not a blocking comment (since it's being done other places in this file), just expressing my displeasure at these one line if-statements with no braces.